src/Microsoft.SqlTools.ServiceLayer/LanguageServices/Completion/SqlCompletionItem.cs

```diff
@@ -45,7 +46,7 @@ private void Init()
         {
             InsertText = GetCompletionItemInsertName();
             Label = DeclarationTitle;
-            if (StartsWithBracket(TokenText))
+            if (StartsWithBracket(TokenText) || IsReservedWord(InsertText))
```

Comment:
>IsReservedWord [](start = 48, length = 14)

have you verified we don't add '[' when autocompleted keywords outside the column or table completion context...e.g., we don't escape SELECT if it's typed by its self.